README


This archive contains the distribution Dist-Zilla-Plugin-MakeMaker-Awesome,
version 0.33:

  A more awesome MakeMaker plugin for L<Dist::Zilla>

This software is copyright (c) 2015 by Ævar Arnfjörð Bjarmason.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.029.



Changes
Revision history for Dist-Zilla-Plugin-MakeMaker-Awesome

0.33      2015-01-03 20:55:21Z
  - re-release to fix problematic $VERSION declaration (RT#101095)

0.32      2014-12-10 02:19:22Z
   - eliminated all uses of Moose::Autobox
   - avoid ever requiring v0 of ExtUtils::MakeMaker

0.31      2014-10-10 05:25:42Z
   - fix failing tests on 5.8.x

0.30      2014-10-09 02:53:56Z
   - use a listref for AUTHOR argument to WriteMakefile(), but only when the
     EUMM version available is already high enough to accomodate it

0.29      2014-10-08 04:26:08Z
   - retain MIN_PERL_VERSION argument to WriteMakefile()
   - extract the minimum perl requirement from all prereq phases, not just
     runtime

0.28      2014-10-02 04:31:44Z
   - fix test failure caused by unstated assumption that
     Dist::Zilla::Role::TestRunner is a very recent version
   - fix uninitialized warning on 5.10.x and earlier

0.27      2014-09-17 03:22:49Z
    - fix test failures (since 0.25) with older versions of [MakeMaker]
    - new 'delimiter' option, which allows the use of leading whitespace in
      'header', 'footer' and 'WriteMakefile_args' strings

0.26      2014-09-16 02:20:28Z
    - new 'preamble' and 'postamble' options renamed to 'header' and 'footer',
      to avoid confusion with MY::preamble and MY::postamble

0.25      2014-09-13 21:17:40Z
    - add new 'preamble' and 'postamble' options, as well as overridable
      builder subs, for inserting code at the top and bottom of Makefile.PL

0.24      2014-08-14 02:07:13Z
    - add x_breaks metadata, and a test that checks it, for
      [MakeMaker::Fallback] versions that we broke with our changes to
      eumm_version handling

0.23      2014-07-30 02:37:52Z
    - create Makefile.PL at filegatherer phase, to allow other plugins to see
      the file earlier

0.22      2014-07-29 04:51:13Z
    - fix for Dist::Zilla::Plugin::MakeMaker 5.020 dropping a default required
      version for ExtUtils::MakeMaker
    - now omitting BUILD_REQUIRES, TEST_REQUIRES in Makefile.PL when empty

0.21      2014-07-28 04:00:35Z
    - *INCOMPATIBLE CHANGE*: the 'test_dirs' sub and '_build_test_dirs' method
      have been renamed to 'test_files' and '_build_test_files' respectively,
      and deal with arrayrefs rather than hashrefs, to more properly reflect
      the format of their values
    - add new 'WriteMakefile_arg', 'test_file', 'exe_file' options, to allow
      for simple use without needing to subclass this plugin

0.20      2014-02-05 05:05:05Z
    - fixed outdated prereq on Dist::Zilla::Plugin::MakeMaker (now requiring
      Dist::Zilla 5 for better prereq merging)

0.19      2014-01-11 18:44:50Z
    - also include the version of ourselves in generated headers
    - synchronized prereqs are used as a fallback in Makefile.PL, as done in
      [MakeMaker] v5.001

0.18      2013-10-20 18:29:23Z
    - re-release with encoding fixes in metadata

0.17      2013-09-27 05:31:27Z
    - fix undef string appearing in place of dist name in generated
      Makefile.PL comment

0.16      2013-09-22 19:48:15Z
    - allow bootstrapping users of this module by using a default $VERSION if
      the real one is not available

0.15      2013-09-08 20:11:17Z
    - re-release with properly encoded META.yml

0.14   2013-05-26 15:12:26
    - test_requires support, as done in Dist::Zilla 4.300032 by Tatsuhiko
      Miyagawa

0.13   2013-05-15 17:11:35
    - fix load issue with Data::Dumper (Karen Etheridge)

0.12   2011-01-29 11:58:27
    - Remove a trailing space in Awesome.pm, patch by Jesse Luehrs at
      https://github.com/avar/dist-zilla-plugin-makemaker-awesome/pull/2

0.11   2010-09-30 10:36:44
    - Remove a trailing space in Awesome.pm, spotted and patched by
      Robin Smidsrød, see
      http://github.com/avar/dist-zilla-plugin-makemaker-awesome/pull/1

0.10   2010-07-02 23:30:25
    - Fixed share_dir breakage caused by Dist::Zilla not being awesome

0.09   2010-07-02 21:51:19
    - Add a DIAGNOSTICS section that notes a common error

0.08   2010-05-30 12:10:40
    - Installing EXE_FILES was broken due to some missed merges in
      0.07.

0.07   2010-05-26 12:17:04
    - Upgrade to depend on Dist::Zilla v3. The required changes to
      Awesome.pm were backported from upstream's MakeMaker.pm.

0.06   2010-05-01 21:18:14
    - Merge changes to the Makefile.PL template from
      Dist::Zilla::Plugin::MakeMaker

0.05   2010-04-09 12:58:41
    - Upgrade to support Dist::Zilla v2
    - Remove copy/paste code we can inherit from
      Dist::Zilla::Plugin::MakeMaker.
    - remove obsolete chainloading support that was never used,
      subclassing is now preferred.

0.04   2010-04-03 12:58:01
    - More documentation proofreading.

0.03   2010-03-31 15:19:08
    - Example from re::engine::Plan9 + more docs

0.02   2010-03-31 14:20:26
    - Added more awesome and renamed to
      Dist::Zilla::Plugin::MakeMaker::Awesome from
      Dist::Zilla::Plugin::OverridableMakeMaker.

0.01   2010-03-13 07:40:07
    - Initial CPAN release


Build.PL
# This Build.PL for Dist-Zilla-Plugin-MakeMaker-Awesome was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-MakeMaker-Awesome was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.006;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.029.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.005"
      },
      "dist_abstract" => "A more awesome MakeMaker plugin for L<Dist::Zilla>",
      "dist_author" => [
        "\x{c6}var Arnfj\x{f6}r\x{f0} Bjarmason <avar\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-MakeMaker-Awesome",
      "dist_version" => "0.33",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::MakeMaker::Awesome",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "CPAN::Meta::Requirements" => "2.121",
        "Dist::Zilla::File::InMemory" => 0,
        "Dist::Zilla::Plugin::MakeMaker" => "5.001",
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::FileGatherer" => 0,
        "List::Util" => 0,
        "Moose" => 0,
        "MooseX::Types::Moose" => 0,
        "MooseX::Types::Stringlike" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.006"
      },
      "script_files" => [],
      "test_requires" => {
        "CPAN::Meta::Check" => "0.007",
        "CPAN::Meta::Requirements" => "2.121",
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::pushd" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => "0.88",
        "if" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      }
    );


    my %fallback_build_requires = (
      "CPAN::Meta::Check" => "0.007",
      "CPAN::Meta::Requirements" => "2.121",
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::pushd" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => "0.88",
      "if" => 0,
      "perl" => "5.006",
      "strict" => 0,
      "warnings" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-MakeMaker-Awesome was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.33.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "A more awesome MakeMaker plugin for L<Dist::Zilla>",
  "AUTHOR" => "\x{c6}var Arnfj\x{f6}r\x{f0} Bjarmason <avar\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-MakeMaker-Awesome",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::MakeMaker::Awesome",
  "PREREQ_PM" => {
    "CPAN::Meta::Requirements" => "2.121",
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => "5.001",
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::FileGatherer" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "MooseX::Types::Moose" => 0,
    "MooseX::Types::Stringlike" => 0,
    "namespace::autoclean" => 0
  },
  "TEST_REQUIRES" => {
    "CPAN::Meta::Check" => "0.007",
    "CPAN::Meta::Requirements" => "2.121",
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.88",
    "if" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.33",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "CPAN::Meta::Check" => "0.007",
  "CPAN::Meta::Requirements" => "2.121",
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Plugin::MakeMaker" => "5.001",
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::FileGatherer" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "List::Util" => 0,
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "MooseX::Types::Moose" => 0,
  "MooseX::Types::Stringlike" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.88",
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
